src/ColumnDisplayFormatDialog.cpp

```diff
@@ -10,29 +10,59 @@ ColumnDisplayFormatDialog::ColumnDisplayFormatDialog(const QString& colname, QSt
+    ui->comboDisplayFormat->insertSeparator(ui->comboDisplayFormat->count());
+    ui->comboDisplayFormat->addItem(tr("Apple NSDate to date"), "appleDate");
+    ui->comboDisplayFormat->addItem(tr("Java epoch (milliseconds) to date"), "javaEpoch");
+    ui->comboDisplayFormat->addItem(tr("Julian day to date"), "julian");
     ui->comboDisplayFormat->addItem(tr("Unix epoch to date"), "epoch");
-    ui->comboDisplayFormat->addItem(tr("Upper case"), "upper");
     ui->comboDisplayFormat->addItem(tr("Windows DATE to date"), "winDate");
+    ui->comboDisplayFormat->insertSeparator(ui->comboDisplayFormat->count());
+    ui->comboDisplayFormat->addItem(tr("Lower case"), "lower");
+    ui->comboDisplayFormat->addItem(tr("Upper case"), "upper");
     ui->labelDisplayFormat->setText(ui->labelDisplayFormat->text().arg(column_name));
 
+    formatFunctions["lower"] = "lower(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["upper"] = "upper(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["epoch"] = "datetime(" + sqlb::escapeIdentifier(column_name) + ", 'unixepoch')";
+    formatFunctions["javaEpoch"] = "strftime('%Y-%m-%d %H:%M:%S.', " + sqlb::escapeIdentifier(column_name) +
+      "/1000, 'unixepoch') || (" + sqlb::escapeIdentifier(column_name) + "%1000)";
+    formatFunctions["winDate"] = "datetime('1899-12-30', " + sqlb::escapeIdentifier(column_name) + " || \" days\")";
+    formatFunctions["appleDate"] = "datetime('2001-01-01', " + sqlb::escapeIdentifier(column_name) + " || \" seconds\")";
+    formatFunctions["julian"] = "datetime(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["round"] = "round(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["hex"] = "printf('0x%x', " + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["octal"] = "printf('%o', " + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["exponent"] = "printf('%e', " + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["hexblob"] = "hex(" + sqlb::escapeIdentifier(column_name) + ")";
+    formatFunctions["decimal"] = "printf('%d', " + sqlb::escapeIdentifier(column_name) + ")";
+
     // Set the current format, if it's empty set the default format
     if(current_format.isEmpty())
     {
         ui->comboDisplayFormat->setCurrentIndex(0);
         updateSqlCode();
     } else {
-        ui->comboDisplayFormat->addItem(tr("Custom"), "custom");
-        ui->comboDisplayFormat->setCurrentIndex(ui->comboDisplayFormat->findData("custom"));
+        QString formatName;
+	foreach (const QString &formatKey, formatFunctions.keys()) {
+	    if(current_format == formatFunctions.value(formatKey)) {
+	        formatName = formatKey;
+		break;
+	    }
+	}
+
+        if(formatName.isEmpty()) {
+	    ui->comboDisplayFormat->insertSeparator(ui->comboDisplayFormat->count());
+            ui->comboDisplayFormat->addItem(tr("Custom"), "custom");
+            formatName = "custom";
+        }
```

Comment:
Just one minor thing: here and above in the foreach loop you're mixing spaces and tabs for indentation which makes the indentation look wrong if the tab width is not set to 8 spaces. Can you change the tabs to 4 spaces instead? :smile: By the way, if you're using Qt Creator you can go to the "Projects" panel (big button on the left), there click "Editor" in the "Project Settings" section, and then set the Tab policy to "Spaces only", the Tab size to "4" and the Indent Size to "4". That changes it just for the sqlitebrowser project and keeps the old settings for all your other projects. That's how I configured it, too :wink:

Reply:
I wasn't aware that Emacs was inserting spaces. I've already changed its settings, so it no longer does that. I don't use Qt Creator for the moment, but I will give it a try.

Reply:
If you're used to Emacs, I'd stick with that (and I say that as a Vim-person :wink:) but Qt Creator is definitely worth giving a try, especially when editing the .ui files :smile:

Reply:
Some tasks are easier with those visual IDEs, but I know too many Emacs tricks and powerful commands for just abandoning it :smile: